File: wrapped/funcs/f_8.py

import re

def is_spam(message):
    # Check for ad-related keywords
    ad_keywords = ['(광고)', '무료거부', '주식', 'http', '추천']
    for keyword in ad_keywords:
        if keyword in message:
            return True

    # Check for excessive use of special characters
    special_chars = ['!', '?', '@', '#', '*', '=', '+']
    count = sum([message.count(char) for char in special_chars])
    if count >= 5:
        return True

    # Check for consecutive capital letters
    capital_letters = re.findall(r'[A-Z|가-힣]{3,}', message)
    if len(capital_letters) >= 2:
        return True

    # Check for numeric sequences
    numeric_sequences = re.findall(r'\d{3,}', message)
    if len(numeric_sequences) >= 1:
        return True

    # If none of the above conditions are met, consider the message as normal
    return False

File: wrapped_kotlin_ported/funcs/f_8.kt

import java.util.regex.Pattern

fun isSpam(message: String): Boolean {
    // Check for ad-related keywords
    val adKeywords = listOf("(광고)", "무료거부", "주식", "http", "추천")
    if (adKeywords.any { keyword -> message.contains(keyword) }) {
        return true
    }

    // Check for excessive use of special characters
    val specialChars = listOf('!', '?', '@', '#', '*', '=', '+')
    val count = specialChars.sumBy { char -> message.count { it == char } }
    if (count >= 5) {
        return true
    }

    // Check for consecutive capital letters
    val pattern = Pattern.compile("[A-Z|가-힣]{3,}")
    val matcher = pattern.matcher(message)
    if (matcher.find() && matcher.find()) {
        return true
    }

    // Check for numeric sequences
    val numericPattern = Pattern.compile("\\d{3,}")
    val numericMatcher = numericPattern.matcher(message)
    if (numericMatcher.find()) {
        return true
    }

    // If none of the above conditions are met, consider the message as normal
    return false
}

